model_api/python/openvino/model_api/tilers/instance_segmentation.py

```diff
@@ -0,0 +1,137 @@
+        self,
+        model,
+        configuration=None,
+        execution_mode="async",
+        tile_classifier_model=None,
+    ):
+        """
+        Constructor for creating a semantic segmentation tiling pipeline
+
+        Args:
+            model: underlying model
+            configuration: it contains values for parameters accepted by specific
+            tiler (`tile_size`, `tiles_overlap` etc.) which are set as data attributes.
+            execution_mode: Controls inference mode of the tiler (`async` or `sync`).
+            tile_classifier_model: an `ImageModel`, which has "tile_prob" output.
+        """
+        super().__init__(model, configuration, execution_mode)
+        self.tile_classifier_model = tile_classifier_model
+
+    def _filter_tiles(self, image, tile_coords, confidence_threshold=0.35):
+        """Filter tiles by objectness score provided by tile classifier
+
+        Args:
+             image: full size image
+             tile_coords: tile coordinates
+
+        Returns:
+             tile coordinates to keep
+        """
+        if self.tile_classifier_model is not None:
+            keep_coords = []
+            for i, coord in enumerate(tile_coords):
+                tile_img = self._crop_tile(image, coord)
+                tile_dict, _ = self.model.preprocess(tile_img)
+                cls_outputs = self.tile_classifier_model.infer_sync(tile_dict)
+                if i == 0 or cls_outputs["tile_prob"] > confidence_threshold:
+                    keep_coords.append(coord)
+            return keep_coords
+
+        return tile_coords
+
+    def _postprocess_tile(self, predictions, meta):
+        """Converts predictions to a format convinient for further merging.
+
+        Args:
+             predictions: predictions from an instance segmentation model: a list of `SegmentedObject` objects
+             meta: a dict containing key "coord", representing tile coordinates
+
+        Returns:
+             a dict with postprocessed detections in 6-items format: (label id, score, bbox) and masks
+        """
+
+        output_dict = super()._postprocess_tile(predictions, meta)
+        if hasattr(predictions, "mask"):
+            output_dict["masks"] = predictions.mask
+        else:
+            output_dict["masks"] = []
+            for segm_res in predictions:
+                output_dict["masks"].append(segm_res.mask)
+
+        return output_dict
+
+    def _merge_results(self, results, shape, meta=None):
+        """Merge results from all tiles.
+
+        To merge detections, per-class NMS is applied.
+
+        Args:
+             results: list of per-tile results
+             shape: original full-res image shape
+        Returns:
+             merged prediciton
+        """
+
+        if meta is None:
+            meta = {}
+        detection_result = super()._merge_results(results, shape, meta)
+
+        masks = []
+        for result in results:
+            if len(result["bboxes"]):
```

Comment:
This should probably be
```suggestion
            if len(result["masks"]):
```

Reply:
Fixed